src/ui/public/agg_types/buckets/geo_hash.js

```diff
@@ -103,25 +109,21 @@ export function AggTypesBucketsGeoHashProvider(Private, config) {
 
       if (agg.params.isFilteredByCollar && agg.getField()) {
         const vis = agg.vis;
-        let mapBounds;
-        let mapZoom;
-        if (vis.hasUiState()) {
-          mapBounds = vis.uiStateVal('mapBounds');
-          mapZoom = vis.uiStateVal('mapZoom');
-        }
+        const mapBounds = vis.sessionState.mapBounds;
+        const mapZoom = getZoomLevel(vis);
         if (mapBounds && mapZoom) {
-          const lastMapCollar = vis.uiStateVal('mapCollar');
+          const lastMapCollar = vis.sessionState.mapCollar;
           let mapCollar;
           if (!lastMapCollar || lastMapCollar.zoom !== mapZoom || isOutsideCollar(mapBounds, lastMapCollar)) {
             mapCollar = scaleBounds(mapBounds);
             mapCollar.zoom = mapZoom;
-            vis.getUiState().set('mapCollar', mapCollar);
           } else {
             mapCollar = lastMapCollar;
           }
           const boundingBox = {};
           delete mapCollar.zoom; // zoom is not part of bounding box filter
           boundingBox[agg.getField().name] = mapCollar;
+          vis.sessionState.mapCollar = mapCollar;
```

Comment:
A new `_msearch` request is created anytime the aggregation changes. By updating the mapCollar every time, the aggregation changes any time the map is moved. This results in an `_msearch` request on each map movement which is not the intended behavior.

`vis.sessionState.mapCollar` should only be updated when the mapBounds are outside of the last fetched map collar and new data needs to be fetched from elasticsearch.